consensus/blake3pow/sealer.go

```diff
@@ -120,22 +120,26 @@ func (blake3pow *Blake3pow) Seal(header *types.WorkObject, results chan<- *types
 	return nil
 }
 
-// mine is the actual proof-of-work miner that searches for a nonce starting from
-// seed that results in correct final header difficulty.
-func (blake3pow *Blake3pow) mine(header *types.WorkObject, id int, seed uint64, abort chan struct{}, found chan *types.WorkObject) {
+func (blake3pow *Blake3pow) Mine(header *types.WorkObject, abort <-chan struct{}, found chan *types.WorkObject) {
 	// Extract some data from the header
 	diff := new(big.Int).Set(header.Difficulty())
 	c, _ := mathutil.BinaryLog(diff, consensus.MantBits)
-	if c <= params.WorkSharesThresholdDiff {
+	workShareThreshold := c - params.WorkSharesThresholdDiff
```

Comment:
is underflow ok here?

Reply:
It is because MineToThreshold guards against underflow